src/org/jetbrains/plugins/scala/lang/refactoring/move/ScalaMoveUtil.scala

```diff
@@ -97,8 +101,15 @@ object ScalaMoveUtil {
       newClass
     }
 
-    if (withCompanion) ScalaPsiUtil.getBaseCompanionModule(aClass).foreach(c => moveClassInner(c, moveDestination))
```

Comment:
The original error came from this line which would create the target file. When moveClassInner was called at the next line, it considered that moving the class to the newly created file would result in creating a fakeCompanion as per `ScalaMoveUtil#canBeCompanions`. moving these lines around wasn't enough though as the expandedElements in `ScalaMoveClassesOrPackagesProcessor` ended up trying to execute the move handler twice in an uncoordinated manner triggering an erroneous fakeCompanion detection again.